### DISCUSSION TITLE
Progress on Rewrite

### DISCUSSION THREAD

**Discussion by AdamLeyshon** [akiradeveloper/sorock#323] [General]
Hello Akira,

I have used this library in a few of our internal services now at [Sigma Financial AI](https://www.sigmafinancial.ai/) to the point where I forked it so I could update Tonic and Prost to more recent versions in line with our own libraries and protocols.

I noticed that not long after that, you began on v2.

Do you have plans to continue updating this library, or should it be considered archived for the foreseeable future?

I'm just trying to determine whether we should stick with this or implement Raft using another library in the short term.

Thanks, Adam.

**Comment by akiradeveloper** [comment]
Nice to meet you Adam.

I understand you forked v1 and maintain the dependencies to make use of it in your internal services. I appreciate your efforts to use lol in your projects. Also, I am bit curious how this library is used in real-world application.

The development of lol has not been updated since the end of Oct. is just because I have been busy after that **not** because I lost interests in it. The fact is that I left Tokyo and moved to a new city which needed me a lot to do in moving.

So the answer to your question is I will continue this project.

> Do you have plans to continue updating this library, or should it be considered archived for the foreseeable future?

The aim of v2 is Multi-raft which allows a server process to include multiple raft applications so they can share the common resource in the process like embedded database or connections. I believe this brings uniqueness to lol. To achieve this, the library's design is sophisticated. Probably, it unintentionally fixed some unnoticed bugs. You will see how big the changes are through #316 and #320.
Another aim of v2 is to reach a stable form that is worth maintaining for years. 

My conclusion is that I recommended you to wait for the release which will be expected in Mar. (possibly Feb.) because I will have a bit of free time in Feb. and Mar. 

At the moment, you can learn the new design from code. In the course of redesigning, some old features are dropped so you'd better check if it keeps the features you wants. Or if you share the code of your internal services with me I will be able to answer quickly.

Sorry for the inconvenience.

**Comment by akiradeveloper** [comment]
done.
